src/Okta.Sdk.Abstractions/DefaultDataStore.cs

```diff
@@ -160,7 +166,18 @@ private void EnsureResponseSuccess(HttpResponse<string> response)
                 throw new InvalidOperationException($"An error occurred deserializing the error body for response code {response.StatusCode}. See the inner exception for details.", ex);
             }
 
-            throw new OktaApiException(response.StatusCode, _resourceFactory.CreateNew<ApiError>(errorData));
+            var contentTypeHeader = response.Headers.FirstOrDefault(x => string.Equals(x.Key, "Content-Type", StringComparison.OrdinalIgnoreCase));
```

Comment:
Is `response.Headers` not some kind of map/dictionary? Could you reference this directly like this: `response.Headers["Content-Type"]`?

Reply:
Unfortunately, you can't do that. You need to create a specific `HttpContent` object based on content-type. For example:
```
case ContentTypeFormUrlEncoded:
                    return string.IsNullOrEmpty(body) ? null : new FormUrlEncodedContent(JsonConvert.DeserializeObject<Dictionary<string, string>>(body));
                default:
                    return string.IsNullOrEmpty(body) ? null : new StringContent(body, Encoding.UTF8, contentType);
```